### DISCUSSION TITLE
What is the replacement for enable_directory_listing == "no"?

### DISCUSSION THREAD

**Discussion by sebhub** [cesanta/mongoose#1517] [Ask questions, get answers] [answered]
In Mongoose 6, there was an `enable_directory_listing` member in `struct mg_serve_http_opts` which could be set to `"no"` to prevent directory listings by `mg_serve_http()`. What can be done in Mongoose 7 to get a similar functionality?

**Comment by cpq** [comment, marked as answer]
It could be enabled / disabled via the build flag, `MG_ENABLE_DIRLIST=0` or `MG_ENABLE_DIRLIST=1`
There is  no corresponding runtime option.

**Reply by sebhub** [reply]
Resurrecting this option would be quite easy and it would restore a feature available in Mongoose 6. The build flag is not a replacement since it disables the directory listing for all directories.

**Reply by linuxguy123** [reply]
That is another good point that it disables dir listing globally, versus letting the developer do it on a situation by situation basis.

**Comment by linuxguy123** [comment]
I think it was a big mistake to remove the run time option to not list the dir.   

It is not reasonable to require the user to rebuild the library with a different build parameter to achieve this.   This means that the developer's functionality can be accidentally over ridden by a user or administrator installing the wrong version of the Mongoose package.
